pkg/types/relayer.go

```diff
@@ -16,12 +16,20 @@ type PluginArgs struct {
 }
 
 type RelayArgs struct {
-	ExternalJobID uuid.UUID
-	JobID         int32
-	ContractID    string
-	New           bool // Whether this is a first time job add.
-	RelayConfig   []byte
-	ProviderType  string
+	ExternalJobID      uuid.UUID
+	JobID              int32
+	ContractID         string
+	New                bool // Whether this is a first time job add.
+	RelayConfig        []byte
+	ProviderType       string
+	MercuryCredentials *MercuryCredentials
```

Comment:
This looks odd to me. Everything else in this struct is at the system level - they are necessary field for any relayer. Whereas this seems very specific.

What is the use case for adding this here?

Reply:
MercuryCredentials can be provided when building the EVM Registry for 2.1 here: https://github.com/smartcontractkit/chainlink/blob/develop/core/services/ocr2/plugins/ocr2keeper/evmregistry/v21/registry.go#L84

In terms of passing additional config, I think we're limited to either relay args or plugin args; is there a more appropriate location where we could put this?

Reply:
were are the mercury credentials stored? are they encrypted in the KeyStore, which is owned by the core node? rather than passing them as a struct here can it be pushed up the stack so that the Relayer has access to an interface that exposes something that can dish out mercury creds?

the evm relayer is already accessing subsets of the key store here https://github.com/smartcontractkit/chainlink/blob/ba841a51666daeb8252d034c5f64da2f637f5f7d/core/services/relay/evm/evm.go#L65C9-L65C9

would extending that work?

Reply:
So I want to say an earlier version of the original PR had modified the keystore interface; but i think that approach fell short at some point! I'll dig into the original PR and see if I can find it! I think I created something like a `CSAETHMercuryKeystore` or something similar 🤔

Reply:
Ah so this is the old PR I'm referring to: https://github.com/smartcontractkit/chainlink/pull/11647/files

Though, looking at those changes, I'm not sure exactly why I abandoned that changeset - I'll try this approach again to see if I can avoid updating the relay args! 👍

Reply:
This comes from the TOML config right? In that case I don't think we are violating anything here.

Reply:
i was assuming that the credentials are not in the TOML. if i'm understanding what you are saying, that sounds like security credential in a text file

Reply:
They are in the `secrets.toml`: https://github.com/smartcontractkit/chainlink/blob/develop/docs/SECRETS.md#mercurycredentialsname

Reply:
@krehermann so the more I look into it; I think the keystore changes I had made previously were addressing a separate feature rather than exposing the mercury credentials; the mercury credentials come from the `DelegateConfig`

Reply:
Ok, thanks for clarifying. Seems like we should merge this then add the ctx everywhere to the interfaces